src/sentry/static/sentry/app/views/dashboardsV2/widgetQueries.tsx

```diff
@@ -164,7 +165,49 @@ class WidgetQueries extends React.Component<Props, State> {
 
           completed++;
           this.setState(prevState => {
-            const tableResults = [...prevState.tableResults, tableData];
+            const tableResults = [...(prevState.tableResults ?? []), tableData];
+            return {
+              ...prevState,
+              tableResults,
+              loading: completed === promises.length ? false : true,
+            };
+          });
+        } catch (err) {
+          const errorMessage =
+            err?.responseJSON?.detail || t('An unknown error occurred.');
+          this.setState({errorMessage});
+        }
+      });
+    } else if (widget.displayType === 'world_map') {
```

Comment:
This method will need a revisit soon its getting long and will have a fair amount of duplication when we add stat charts.